### PR TITLE
fix: stop zeebe task completion variables from being serialized

### PR DESCRIPTION
**Pull request by panagiotisgts**
## Description

Address tasklist issue mentioned [here](https://camunda.slack.com/archives/C046XS3GGCR/p1724945281241419?thread_ts=1724936257.713079&cid=C046XS3GGCR): Completed Zeebe User Task form variables were being flushed as serialized JSON values in ES/OS.

### CODE REVIEW

**Comment by nathansandi** [comment]
The importers were just updated with the list-view setup,

Is this change necessary on the list view requests as well?

**Comment by houssain-barouni** [comment]
@panagiotisgts before `Json.createValue` we were previously using org.json.JSONObject(string) that transforms primitive types (booleans, numbers, nulls) to their respective types
We have removed the use of org.json.JSONObject in https://github.com/camunda/camunda/pull/19279  but probably not with the right implementation
can you confirm that it works with native types ? for example `"true"` should not be replaced with `"\"true\""`

**Comment by panagiotisgts** [comment]
Hey @houssain-barouni, run over a form to provide the values for cross checking, I believe we are good, if you can verify this also.  This is the view from the `tasklist-task-variable` index regarding a set of variables present on the form
image

And this is the response of the API for `v1/tasks/{uuid}/variables/search`
```json
[ {
  "id" : "2251799813685272-checkbox_pw88bp",
  "name" : "checkbox_pw88bp",
  "value" : "\"false\"",
  "isValueTruncated" : false,
  "previewValue" : "\"false\"",
  "draft" : null
}, {
  "id" : "2251799813685272-checkbox_zvnb6",
  "name" : "checkbox_zvnb6",
  "value" : "\"true\"",
  "isValueTruncated" : false,
  "previewValue" : "\"true\"",
  "draft" : null
}, {
  "id" : "2251799813685272-datetime_zy5lc",
  "name" : "datetime_zy5lc",
  "value" : "\"2024-09-12\"",
  "isValueTruncated" : false,
  "previewValue" : "\"2024-09-12\"",
  "draft" : null
}, {
  "id" : "2251799813685272-radio_f3pz6n",
  "name" : "radio_f3pz6n",
  "value" : "\"value2\"",
  "isValueTruncated" : false,
  "previewValue" : "\"value2\"",
  "draft" : null
}, {
  "id" : "2251799813685272-textfield_first",
  "name" : "textfield_first",
  "value" : "\"testField1\"",
  "isValueTruncated" : false,
  "previewValue" : "\"testField1\"",
  "draft" : null
}, {
  "id" : "2251799813685272-textfield_second",
  "name" : "textfield_second",
  "value" : "\"123\"",
  "isValueTruncated" : false,
  "previewValue" : "\"123\"",
  "draft" : null
} ]
```

**Comment by houssain-barouni** [comment]
> Hey @houssain-barouni, run over a form to provide the values for cross checking, I believe we are good, if you can verify this also. This is the view from the `tasklist-task-variable` index regarding a set of variables present on the form image
> 
> And this is the response of the API for `v1/tasks/{uuid}/variables/search`
> 
> ```json
> [ {
>   "id" : "2251799813685272-checkbox_pw88bp",
>   "name" : "checkbox_pw88bp",
>   "value" : "\"false\"",
>   "isValueTruncated" : false,
>   "previewValue" : "\"false\"",
>   "draft" : null
> }, {
>   "id" : "2251799813685272-checkbox_zvnb6",
>   "name" : "checkbox_zvnb6",
>   "value" : "\"true\"",
>   "isValueTruncated" : false,
>   "previewValue" : "\"true\"",
>   "draft" : null
> }, {
>   "id" : "2251799813685272-datetime_zy5lc",
>   "name" : "datetime_zy5lc",
>   "value" : "\"2024-09-12\"",
>   "isValueTruncated" : false,
>   "previewValue" : "\"2024-09-12\"",
>   "draft" : null
> }, {
>   "id" : "2251799813685272-radio_f3pz6n",
>   "name" : "radio_f3pz6n",
>   "value" : "\"value2\"",
>   "isValueTruncated" : false,
>   "previewValue" : "\"value2\"",
>   "draft" : null
> }, {
>   "id" : "2251799813685272-textfield_first",
>   "name" : "textfield_first",
>   "value" : "\"testField1\"",
>   "isValueTruncated" : false,
>   "previewValue" : "\"testField1\"",
>   "draft" : null
> }, {
>   "id" : "2251799813685272-textfield_second",
>   "name" : "textfield_second",
>   "value" : "\"123\"",
>   "isValueTruncated" : false,
>   "previewValue" : "\"123\"",
>   "draft" : null
> } ]
> ```

that's actually the problem , `true`, `false`, `123`, should be stored in ES without the double quotes
and they should be returned by the API as `"true"`, `"false"`, `"123"`

**Comment by panagiotisgts** [comment]
@houssain-barouni Thanks for the feedback, I can confirm that for `Boolean` types, they are not stored with quotes ("true") for JobWorker User Tasks, but regarding `Number` values they are stored quoted for JobWorkers as well.

**Review by nathansandi — APPROVED** [review]
Great work here 💯

**Comment by houssain-barouni** [comment]
> @houssain-barouni Thanks for the feedback, I can confirm that for `Boolean` types, they are not stored with quotes ("true") for JobWorker User Tasks, but regarding `Number` values they are stored quoted for JobWorkers as well.

I don't get it, maybe we can have a call

**Review by houssain-barouni — COMMENTED** [review]
@panagiotisgts I need a sync call to walk me through the Mapper that removes/adds the double quotes

File: tasklist/importer-860/src/main/java/io/camunda/tasklist/zeebeimport/v860/processors/common/UserTaskRecordToVariableEntityMapper.java [hunk 1]
```diff
@@ -36,15 +40,19 @@ public List<TaskVariableEntity> mapVariables(final Record<UserTaskRecordValue> r
...
       final Map<String, Object> variablesMap = recordValue.getVariables();
       for (final Map.Entry<String, Object> varMap : variablesMap.entrySet()) {
-        final String varValue = String.valueOf(varMap.getValue());
-
+        final String varValue;
+        try {
+          varValue = objectMapper.writeValueAsString(varMap.getValue());
+        } catch (final JsonProcessingException e) {
+          LOGGER.error("Failed to parse variable %s".formatted(varMap), e);
+          throw new RuntimeException(e);
```
Comment (nathansandi) [inline tasklist/importer-860/src/main/java/io/camunda/tasklist/zeebeimport/v860/processors/common/UserTaskRecordToVariableEntityMapper.java:48]: Lets use the Tasklist Custom Exception here  
'TasklistRuntimeException'
over here

File: tasklist/importer-860/src/main/java/io/camunda/tasklist/zeebeimport/v860/processors/es/UserTaskZeebeRecordProcessorElasticSearch.java [hunk 2]
```diff
@@ -108,16 +107,8 @@ private UpdateRequest getVariableQuery(final TaskVariableEntity variable)
...
-          TaskVariableTemplate.VALUE,
-          "null".equals(variable.getValue())
-              ? "null"
-              : objectMapper.writeValueAsString(Json.createValue(variable.getValue())));
-      updateFields.put(
-          TaskVariableTemplate.FULL_VALUE,
-          "null".equals(variable.getFullValue())
-              ? "null"
-              : objectMapper.writeValueAsString(Json.createValue(variable.getFullValue())));
+      updateFields.put(TaskVariableTemplate.VALUE, variable.getValue());
```
Comment (nathansandi) [inline tasklist/importer-860/src/main/java/io/camunda/tasklist/zeebeimport/v860/processors/es/UserTaskZeebeRecordProcessorElasticSearch.java:110]: Did you test the situation if the`variable`comes null? 
Is this not gonna throw a null pointer exception?

**Review by nathansandi — CHANGES_REQUESTED** [review]
Nice work here, I have some questions in order to understand more this fix: 
If we dont save anymore the values with `\"`

- Does will impact in API Search calls? For example when we search a Task by Variable Name/Value?

- Did you check if this impacts on this functionality from Search Task in V1/V2? Or the variables are saved in the same way

Related t the changes, just one minor

**Comment by panagiotisgts** [comment]
> * Does will impact in API Search calls? For example when we search a Task by Variable Name/Value?
> * Did you check if this impacts on this functionality from Search Task in V1/V2? Or the variables are saved in the same way

Since it's the same way that Job Worker variables are being persisted it should cause no issue. String variables are still stored wrapped in quotes.

**Review by nathansandi — APPROVED** [review]
LGMT, nice changes on test.

**Pull request merged**